project1/src/components/OnSale.tsx

```diff
@@ -0,0 +1,126 @@
+"use client";
+
+import ProductCard from "./ProductCard";
+import { Product } from "../types/Product";
+import { useEffect, useState } from "react";
+import { getPromotions } from "@/services/ProductService";
+import { FaChevronLeft, FaChevronRight } from "react-icons/fa";
+import { Button } from "@/components/ui/button";
+
+export default function OnSale() {
+  const [products, setProducts] = useState<Product[]>([]);
+  const [loading, setLoading] = useState(true);
+  const [currentPage, setCurrentPage] = useState(0);
+  // Responsive items per page: mobile 4, tablet 6, desktop 6
+  const [itemsPerPage, setItemsPerPage] = useState(4);
+
+  useEffect(() => {
+    getPromotions()
+      .then(setProducts)
+      .catch(console.error)
+      .finally(() => setLoading(false));
+  }, []);
+
+  // Update items per page based on screen size
+  useEffect(() => {
+    const updateItemsPerPage = () => {
+      if (window.innerWidth < 640) {
+        setItemsPerPage(4); // Mobile: 2 columns x 2 rows
+      } else if (window.innerWidth < 1024) {
+        setItemsPerPage(6); // Tablet: 2 columns x 3 rows
+      } else {
+        setItemsPerPage(6); // Desktop: 3 columns x 2 rows
+      }
+    };
```

Comment:
logic này mình làm gọn hơn được e nhé
```suggestion
     setItemsPerPage(window.innerWidth < 640 ? 4 : 6)
```